src/main/java/org/broadinstitute/hellbender/tools/spark/transforms/markduplicates/MarkDuplicatesSparkUtils.java

```diff
@@ -347,58 +454,59 @@ public static void saveMetricsRDD(final MetricsFile<DuplicationMetrics, Double>
     }
 
     /**
-     * GATKRead comparator that compares based on mapping position followed by SAM flags.
+     * Comparator for sorting Reads by coordinate. Note that a header is required in
+     * order to meaningfully compare contigs.
+     *
+     * Uses the various other fields in a read to break ties for reads that share
+     * the same location.
+     *
+     * Ordering is almost identical to the {@link htsjdk.samtools.SAMRecordCoordinateComparator},
+     * modulo a few subtle differences in tie-breaking rules for reads that share the same
+     * position. This comparator will produce an ordering consistent with coordinate ordering
+     * in a bam file, including interleaving unmapped reads assigned the positions of their
+     * mates with the mapped reads.
      */
-    final static class GATKOrder implements Comparator<GATKRead>, Serializable {
-        private static final long serialVersionUID = 1l;
-        private final SAMFileHeader header;
-        // TODO: Unify with other comparators in the codebase
+    public static final class PairedEndsCoordinateComparator implements Comparator<PairedEnds>, Serializable {
```

Comment:
It would probably be better to have this comparator live in a standalone class in the same package instead of embedded in this utils class like this.